narwhals/_spark_like/expr.py

```diff
@@ -560,6 +562,65 @@ def _len(_input: Column) -> Column:
 
         return self._with_callable(_len)
 
+    def replace_strict(
+        self,
+        old: Sequence[Any] | Mapping[Any, Any],
+        new: Sequence[Any],
+        *,
+        return_dtype: DType | type[DType] | None,
+    ) -> Self:
+        mapping = old if isinstance(old, Mapping) else dict(zip(old, new))
+
+        mapping_keys = list(mapping.keys())
+
+        # Create an array of all valid keys for our IN check
+        # Note: None/null handling is special in Spark - we'll handle it separately
+        non_null_keys = [k for k in mapping_keys if k is not None]
+        has_null_key = None in mapping_keys
+
+        mapping_expr = self._F.create_map(
+            [self._F.lit(x) for x in chain(*mapping.items())]
+        )
+
+        def _replace_strict(_input: Column) -> Column:
+            validation_expr = (
+                self._F.when(
+                    _input.isNull() & self._F.lit(has_null_key),
+                    self._F.lit(True),  # noqa: FBT003
+                )
+                .when(_input.isNull() & ~self._F.lit(has_null_key), self._F.lit(False))  # noqa: FBT003
+                .otherwise(
+                    self._F.array_contains(
+                        self._F.array([self._F.lit(k) for k in non_null_keys]), _input
+                    )
+                )
+            )
+
+            mapped_col = (
+                mapping_expr[_input]
+                if self._implementation.is_pyspark()
+                else mapping_expr.getItem(_input)
+            )
```

Comment:
PySpark raises a warning for `mapping_expr.getItem(_input)`, while SQLFrame raises an exception for `mapping_expr[_input]` (Column is not callable)